### PR TITLE
Feature/post report

### LINKED ISSUE #132: Cannot like a post

### PR DESCRIPTION
**Pull request by xavier-campos-diaz**

### CODE REVIEW

**Review by enrichp — DISMISSED** [review]

**Review by enrichp — DISMISSED** [review]

**Review by marclll — DISMISSED** [review]

**Review by enrichp — DISMISSED** [review]

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    private Post post;
+    private boolean result;
+
+    public TrUnbanPost(CommunityService communityService) {
+        this.communityService = communityService;
+    }
+
+    /**
+     * Setter of the user that wants to unban the post.
+     * @param user
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java:23]: Codacy Issue found: [At-clause should have a non-empty description.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 14
           

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405&bid=18477393)

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     */
+    public void setUser(User user) {
+        this.user = user;
+    }
+
+    /**
+     * Setter of the post that has to be unbanned
+     * @param post
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java:31]: Codacy Issue found: [At-clause should have a non-empty description.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java [hunk 3]
```diff
@@ -203,8 +203,7 @@ public void deletePetsFromUser(User user) {
         ExecutorService executorService = Executors.newSingleThreadExecutor();
         executorService.execute(() -> {
             try {
-                userPets.set(ServiceLocator.getInstance().getPetManagerClient().getAllPets(user.getToken(),
-                        user.getUsername()));
+                userPets.set(ServiceLocator.getInstance().getPetManagerClient().getAllPets(user.getToken(), user.getUsername()));
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java:206]: Codacy Issue found: [Line is longer than 120 characters (found 129).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/medication/MedicationManagerAdapter.java [hunk 4]
```diff
@@ -89,7 +89,7 @@ public void updateMedicationKey(User user, Pet pet, String newDate, String oldDa
             MedicationData medicationData = null;
             try {
                 medicationData = ServiceLocator.getInstance().getPetCollectionsManagerClient()
-                        .getMedication(user.getToken(), user.getUsername(), pet.getName(), oldDate);
+                        .getMedication(user.getToken(), user.getUsername() ,pet.getName(), oldDate);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/medication/MedicationManagerAdapter.java:92]: Codacy Issue found: [',' is not followed by whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/community/CommunityAdapter.java [hunk 5]
```diff
@@ -406,7 +412,17 @@ public void unlikePost(User user, Post post) {
 
     @Override
     public void reportPost(User user, Post post, String reportMessage) {
-        // Not implemented yet
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        executorService.execute(() -> {
+            try {
+                ServiceLocator.getInstance().getForumManagerClient().reportMessage(user.getToken(),
+                        post.getForum().getGroup().getName(), post.getForum().getName(), post.getUsername(), user.getUsername(),
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/community/CommunityAdapter.java:419]: Codacy Issue found: [Line is longer than 120 characters (found 128).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * Setter of the user that wants to unban the post.
+     * @param user
+     */
+    public void setUser(User user) {
+        this.user = user;
+    }
+
+    /**
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java:29]: Codacy Issue found: [First sentence should end with a period.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java [hunk 7]
```diff
@@ -273,6 +272,16 @@ private void initializeButtons(CircularEntryView circularEntryView, View options
...
         });
+
+        User user = InfoGroupFragment.getCommunication().getUser();
+        if (selectedPost.isBanned() && forum.getOwnerUsername().equals(user.getUsername())) {
+            MaterialButton btnUnban = optionsPostLayout.findViewById(R.id.unbanPostButton);
+            btnUnban.setOnClickListener(v -> {
+                InfoGroupFragment.getCommunication().unbanPost(selectedPost);
+            });
+        } else {
+            optionsPostLayout.findViewById(R.id.unbanPostButton).setVisibility(View.GONE);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java:283]: Codacy Issue found: [Line matches the illegal pattern 'Use ViewBinding instead'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/medication/MedicationManagerAdapter.java (hunk 4 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/medication/MedicationManagerAdapter.java:92]: Codacy Issue found: [',' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java [hunk 8]
```diff
@@ -243,8 +241,7 @@ public void deletePetsFromUser(User user) {
         ExecutorService executorService = Executors.newSingleThreadExecutor();
         executorService.execute(() -> {
             try {
-                pets.set(ServiceLocator.getInstance().getPetManagerClient().downloadAllProfileImages(user.getToken(),
-                        user.getUsername()));
+                pets.set(ServiceLocator.getInstance().getPetManagerClient().downloadAllProfileImages(user.getToken(), user.getUsername()));
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java:244]: Codacy Issue found: [Line is longer than 120 characters (found 139).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java [hunk 9]
```diff
@@ -94,8 +94,8 @@ public boolean registerNewPet(User user, Pet pet) {
             Pair<DateTime, Double> entry = pet.getLastWeightInfo();
             Weight libraryWeight = new Weight(entry.first.toString(), entry.second);
             try {
-                ServiceLocator.getInstance().getPetManagerClient().addFieldCollectionElement(user.getToken(),
-                    user.getUsername(), pet.getName(), PetData.WEIGHTS, libraryWeight.getKey(),
+                ServiceLocator.getInstance().getPetManagerClient().addFieldCollectionElement(user.getToken(), user.getUsername(),
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java:97]: Codacy Issue found: [Line is longer than 120 characters (found 129).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java [hunk 10]
```diff
@@ -273,6 +272,16 @@ private void initializeButtons(CircularEntryView circularEntryView, View options
         btnReport.setOnClickListener(v -> {
             addReportButtonListener(circularEntryView, editPostDialog);
         });
+
+        User user = InfoGroupFragment.getCommunication().getUser();
+        if (selectedPost.isBanned() && forum.getOwnerUsername().equals(user.getUsername())) {
+            MaterialButton btnUnban = optionsPostLayout.findViewById(R.id.unbanPostButton);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java:278]: Codacy Issue found: [Line matches the illegal pattern 'Use ViewBinding instead'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/user/UserManagerAdapter.java [hunk 11]
```diff
@@ -105,8 +104,7 @@ public boolean userExists(User user) {
         ExecutorService executorService = Executors.newSingleThreadExecutor();
         executorService.execute(() -> {
             try {
-                userData.set(ServiceLocator.getInstance().getUserManagerClient().getUser(user.getToken(),
-                        user.getUsername()));
+                userData.set(ServiceLocator.getInstance().getUserManagerClient().getUser(user.getToken(), user.getUsername()));
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/user/UserManagerAdapter.java:107]: Codacy Issue found: [Line is longer than 120 characters (found 127).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+     * Executes the transaction.
+     * @throws NotForumOwnerException The given user is not the owner of the forum were the post is made
+     */
+    public void execute() throws NotForumOwnerException {
+        if (post.getForum().getOwnerUsername().equals(user.getUsername())) {
+            this.communityService.unbanPost(user, post);
+        }
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnbanPost.java:52]: Codacy Issue found: ['}' at column 9 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java [hunk 13]
```diff
@@ -328,8 +325,8 @@ public void deletePetWeight(User user, Pet pet, DateTime dateTime) {
         ExecutorService executorService = Executors.newSingleThreadExecutor();
         executorService.execute(() -> {
             try {
-                ServiceLocator.getInstance().getPetManagerClient().deleteFieldCollectionElement(user.getToken(),
-                        user.getUsername(), pet.getName(), PetData.WEIGHTS, dateTime.toString());
+                ServiceLocator.getInstance().getPetManagerClient().deleteFieldCollectionElement(user.getToken(), user.getUsername(),
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java:328]: Codacy Issue found: [Line is longer than 120 characters (found 132).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5643405)

**Review by marclll — APPROVED** [review]

**Review by santidrj — APPROVED** [review]
:+1:

**Review by AlbertPG — APPROVED** [review]
Okey

**Pull request merged**